hudi-flink/src/main/java/org/apache/hudi/sink/partitioner/profile/WriteProfile.java

```diff
@@ -101,13 +105,19 @@
    */
   private final Configuration hadoopConf;
 
+  /**
+   * Metadata cache to reduce IO of metadata files.
+   */
+  private final Map<String, HoodieCommitMetadata> metadataCache;
+
   public WriteProfile(HoodieWriteConfig config, HoodieFlinkEngineContext context) {
     this.config = config;
     this.basePath = new Path(config.getBasePath());
     this.smallFilesMap = new HashMap<>();
     this.recordsPerBucket = config.getCopyOnWriteInsertSplitSize();
     this.table = HoodieFlinkTable.create(config, context);
     this.hadoopConf = StreamerUtil.getHadoopConf();
+    this.metadataCache = new HashMap<>();
```

Comment:
Hi, @danny0405 , Thank you for your contribution.
I have a little confusion about that caches usually have a fixed capacity (number or size), is a similar specification needed here?  When the capacity is exceeded, cleanup will be performed according to a fixed policy.

Reply:
This cache did have limit for cache size, see `cleanMetadataCache`, it actually has the max number of metadata items as the active time lime limit, by default is 30.